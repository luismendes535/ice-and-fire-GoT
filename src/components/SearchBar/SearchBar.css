.SearchBar {
  display: flex;
  align-items: center;
}
.Input {
  margin-right: 1rem;
  box-shadow: 4px 3px 20px 0px rgba(104, 103, 107, 1);
  border-radius: 3px;
  padding: 5px;
}

.Checkbox {
  margin-right: 1rem;
  cursor: pointer;
}

.Button {
  margin-left: 1rem; 
  box-shadow: 4px 3px 20px 0px rgba(104, 103, 107, 1);
  padding: 0.5rem 1rem;
  background-color: white;
  border-radius: 5px;
  cursor: pointer;
}
