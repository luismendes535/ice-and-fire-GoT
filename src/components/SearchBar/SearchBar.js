import React from "react";
import "./SearchBar.css";
import PropTypes from "prop-types";

const SearchBar = ({
  searchedChar,
  isAlive,
  updateSearchInput,
  toggleAliveChars,
  onSearchClick
}) => {
  return (
    <div className="SearchBar">
      <input
        className="Input"
        value={searchedChar}
        placeholder="Character name"
        onChange={e => updateSearchInput(e)}
      />
      <p>
        <input
          className="Checkbox"
          checked={isAlive}
          type="checkbox"
          onClick={() => toggleAliveChars()}
        />
        Is alive
      </p>
      <button className="Button" onClick={() => onSearchClick()}>
        Search
      </button>
    </div>
  );
};

SearchBar.propTypes = {
  searchedChar: PropTypes.func,
  isAlive: PropTypes.bool,
  updateSearchInput: PropTypes.func,
  toggleAliveChars: PropTypes.func,
  onSearchClick: PropTypes.func
};

export default SearchBar;
