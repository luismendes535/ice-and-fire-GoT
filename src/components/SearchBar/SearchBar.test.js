import React from "react";
import { shallow } from "enzyme";
import SearchBar from "./SearchBar";

describe("<SearchBar />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchBar />);
  });
  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should render <SearchBar/>",()=>{
      expect(wrapper).toHaveLength(1)
  })
});
