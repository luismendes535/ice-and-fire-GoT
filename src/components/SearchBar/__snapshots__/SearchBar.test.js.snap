// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render properly 1`] = `ShallowWrapper {}`;
